src/content_sao.cpp

```diff
@@ -1440,6 +1448,23 @@ void PlayerSAO::setWieldIndex(int i)
 	}
 }
 
+ItemStack PlayerSAO::getOldWieldedItem() const
+{
+	return m_old_wielded_item;
+}
+
+void PlayerSAO::setOldWieldedItem(const ItemStack &item)
+{
+	m_old_wielded_item = item;
+}
+
+void PlayerSAO::resetWielditemTimer()
+{
+	m_wielditem_interval = IntervalLimiter();
+	m_wielditem_timer_active = true;
+	infostream << "PlayerSAO::resetWielditemTimer()" << std::endl;
```

Comment:
I don't quite see what this timer is for

Reply:
The timer is for ensuring that the on_wielditem_change callbacks are run only if the item has been selected for more than 0.5s. This prevents the callback from being invoked for every item when scrolling from say, slot 1 to slot 8. The timer_active boolean exists to make the timer run only once after a wielditem change, to prevent the callback from being run every 0.5s.